tools/snippet-testing/language_handler/java.rb

```diff
@@ -20,7 +20,7 @@ def text_with_specific_replacements(file_content)
     end
 
     def text_with_example_class_name(file_content)
-      file_content.gsub(/^public class\s.+\s*{\s*$/, "public class #{TEST_CLASS_NAME} {")
+      file_content.gsub(/^public class\s[^\s{]+/, "public class #{TEST_CLASS_NAME}")
```

Comment:
@hortega could you check if this works?

```rb
file_content.gsub(/(?<=^public class).*(?={\s*$)/, " #{TEST_CLASS_NAME} ")
```

Also I think we should allow the regex to match patterns like

```diff
+ public class X {              // notice the spaces
```

As not all contributors auto removes trailing spaces after `{`, wdyt?